Improve structure of the tests

suutari-ai opened the pull request:
This makes bare "pytest" work for running the tests.

The test settings are now in a separate file, so they are more cleanly
separated (and not tied to a runner script).

There is also some unrelated fixes to tox.ini.

There is no real need for a wrapper script anymore, but a bigger project
might need one later, so therefore I added a bare runtests script that
runs tox.  More functionality can be added to that later, if something
is required.

suutari-ai committed:
Improve structure of the tests

This makes bare "pytest" work for running the tests.

The test settings are now in a separate file, so they are more cleanly
separated (and not tied to a runner script).

There is also some unrelated fixes to tox.ini.

There is no real need for a wrapper script anymore, but a bigger project
might need one later, so therefore I added a bare runtests script that
runs tox.  More functionality can be added to that later, if something
is required.

suutari-ai commented:
Btw., check the diffstat "+53 −101". That's also one reason why I prefer this approach over the old one. I.e. less code. :)

frwickst commented on {{cookiecutter.project_slug}}/runtests:
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+set -e
+tox "$@"
```
Why should runtests run tox? As I mentioned when we talked earlier, I see tox as something that wraps the tests and runs them with different python versions or packages. Tox should be the one running this file imho.
Like you said yourself, it should be fast to run the tests locally, this will be slow since it tries to run all the tox tests. Also, tox most likely defines several python versions that the developer does not have installed locally, so the tests will fail due to missing Python versions.

Tox.ini should run `runtests` not the other way around and runtests should run py.tests. At least imho.

JsseL commented:
Seems reasonable and good changes to me, but the purpose of `runtests` indeed feels unclear. Should there be a mention about it in the README or in file comments stating the purpose of it?

frwickst commented:
Another way is also to take it one step further and include a Makefile :) Then everything can be done with one command and arguments to that. For instance:
`make build`: Builds the package
`make tests`: Tests the package
`make docs`: Generates documentation
`make clean`: Cleans up

The runtests.py could still be in there, the Makefile would just call it.

I do agree that now the runtests is empty and does not really serve any purpose if it is made like this.  Might as well be removed in this case. Can always be added when needed. What I kinda liked about it, or then the `make` approach above is that there is no need to know if we are using py.test of normal Django tests or something else, the command would always be the same.

suutari-ai commented:
I like the `Makefile` idea very much.

Also agree that it would be nice if there always some some "common entrypoint" for running the tests so that the command would always be the same. IMHO that could be `make test`.

The `runtests` script isn't really needed here and should probably be removed. Like @frwickst said, it can be introduced to a project if there is a need for it.

I'll update the PR when I find time. Ping me if you want to work on this instead or if there is some project that will need this soon(er).

suutari-ai commented on {{cookiecutter.project_slug}}/setup.cfg:
```diff
@@ -32,14 +32,12 @@ include_package_data = True
 
 [tool:pytest]
 norecursedirs = .* build dist venv test_data
+DJANGO_SETTINGS_MODULE = {{cookiecutter.project_name}}.tests.settings
```
This should use `project_slug` instead. (Noticed while was testing this change.)

suutari-ai commented:
This is still missing the post generate script that removes the Django files for non-Django projects.

suutari-ai committed:
Improve flake8 and isort configuration

suutari-ai committed:
Rename requirements-stylecheck to requirements-style

These deps are not only for checking the style, but also help fixing the
style, e.g. "isort" is installed with these tools and it can be used for
fixing the imports.

suutari-ai committed:
Fix README name in MANIFEST.in

suutari-ai committed:
Rename maintainer to author in setup.cfg

suutari-ai committed:
Fix author and email field usage

suutari-ai committed:
Allow configuring project_url and keywords

suutari-ai committed:
Rename the top-level directory to use dashes

suutari-ai committed:
Add hook for removing Django specific tests dir

suutari-ai committed:
Fix title underlining length in README.rst

suutari-ai committed:
Fix empty first line on generated LICENSE

suutari-ai committed:
Fix django settings

Add SECRET_KEY as its required.  Fix style issue with TEMPLATES.  Move
DEBUG to top.

suutari-ai commented:
@frwickst @JsseL Updated this now.

JsseL commented on hooks/post_gen_project.py:
```diff
@@ -0,0 +1,25 @@
+    underlining = (len(first_line.rstrip()) * '=') + '\n'
+    new_lines = [first_line, underlining] + list(lines)
+with io.open('README.rst', 'wt', encoding='utf-8') as fp:
+    fp.write(''.join(new_lines))
```
Why not.. IMO totally unnecessary trickery though 😄

JsseL commented:
Seems fine to me. 👍 :shipit:

frwickst commented:
![image](https://user-images.githubusercontent.com/3242350/38303800-bbcfe30a-380f-11e8-838f-b311ef653381.png)
